src/goto-programs/goto_program.cpp

```diff
@@ -532,10 +533,33 @@ std::string as_string(
 /// order they appear in the goto program.
 void goto_programt::compute_loop_numbers()
 {
+  // This deals with nested loop number assignment by treating the queue
+  // as a loop stack, and it'll hold whatever number is currently considered
+  // active.
   unsigned nr=0;
+  std::queue<int> active_loops;
   for(auto &i : instructions)
+  {
+    bool loop_target = false;
+    for(auto &inc_edge : i.incoming_edges)
+      loop_target = loop_target || inc_edge->is_backwards_goto();
```

Comment:
Check that number of incoming edges is at least 2 as well, otherwise we're looking at an ordinary `goto` instruction